### DISCUSSION TITLE
What are the breaking changes in 8.0.x?

### DISCUSSION THREAD

**Discussion by NxSoftware** [reactiveui/refit#1931] [Q&A] [answered]
According to the main [README.md](https://github.com/reactiveui/refit/blob/main/README.md#updates-in-80x) for updates in 8.0.x
> this lead to some breaking changes See [Releases](https://github.com/reactiveui/refit/releases) for full details

However the releases page does not list any breaking changes for 8.0.0. Are there actually any breaking changes and if so what are they?

**Comment by TimothyMakkison** [comment]
Iirc the major changes are:
- Prevent CRLF injection in headers
- Switching to an incremental generator
- Changed `try catch` to give a more detailed stack trace

Reason given here: https://github.com/reactiveui/refit/pull/1864#issuecomment-2395613127

**Comment by anaisbetts** [comment]
The short version is, "There aren't really any" 😅

**Comment by glennawatson** [comment, marked as answer]
8.0.x is effectively a non-breaking release for normal usage; the README wording is overcautious. The notable changes are:

- The Refit source generator moved to an incremental generator. No API change, but it requires a recent build toolchain (modern Roslyn / .NET SDK).
- CRLF injection protection was added for header values. If you were (intentionally or not) putting control characters in headers, those are now sanitized/rejected.
- Internal try/catch handling was reworked to surface more detailed stack traces on errors.

If your project builds with a current SDK and you are not injecting raw control characters into headers, upgrading from 7.x to 8.0.x should require no code changes. Context: https://github.com/reactiveui/refit/pull/1864#issuecomment-2395613127
